lms/djangoapps/courseware/middleware.py

```diff
@@ -35,3 +38,19 @@ def process_request(self, request):
 
             if course_id and course_id != request.session.get('course_id'):
                 request.session['course_id'] = course_id
+
+
+class BlockingUrlMiddleware(object):
```

Comment:
Looking good now but it would be better if we could have this middleware separate in the redhouse app. We avoid making changes in the core as much as possible.